### PR TITLE
Fix stuck bots 3.0

### PR DESCRIPTION
**Pull request by necessarily-equal**
This makes bot try harder to move even when they aren't supposed to be able to avoid the obstacle, which makes them less stuck.

This version makes them try to run a bit sideway, one time left, one time right until they find their way. It seems more efficient but also a bit more complex. See also #2214 for the simpler version.

Fix https://github.com/Unvanquished/Unvanquished/issues/2182.

### CODE REVIEW

File: src/sgame/sg_bot_nav.cpp [hunk 1]
```diff
@@ -598,10 +598,25 @@ bool BotMoveToGoal( gentity_t *self )
...
+
+		int clientNum = self - g_entities;
+		dir[2] = 0;
+		dir = glm::normalize(dir);
+		glm::vec3 up = { 0, 0, 1 };
+		glm::vec3 right = glm::cross(dir, up);
+
+		// invert periodically four times per second. We add a delta
+		// determined from clientNum to avoid bots dancing in sync
+		bool invert = ((110 * clientNum + self->client->time1000) % 125) < 250;
```
Comment (slipher) [inline src/sgame/sg_bot_nav.cpp:611]: The condition is always true, barring integer overflows. Did you test this?

Comment (necessarily-equal) [reply src/sgame/sg_bot_nav.cpp]: Good catch!

Bots still moved randomly by changing target, that made it look about right :eyes:. Fixed. I think it looks better now.

**Comment by bmorel** [comment]
This hack should not go in this function, but in BotFindSteerTarget(), which already tries to do what you try to do, or in the unstuck (BT) routine.

File: src/sgame/sg_bot_nav.cpp [hunk 2]
```diff
@@ -412,9 +412,38 @@ static bool BotShouldJump( gentity_t *self, const gentity_t *blocker, const glm:
...
+
+	// 280 instead of 700/2=350 because we want some left/right
+	// asymetry in case it helps
+	bool invert = time < 280;
+	// half of the time we go straight instead
+	bool sideway = time > 700;
+
+	// 2*right + 1*dir means an angle of atan(2/1) = 63°. This
+	// seems about right
+	dir += right * (invert ? -1.0f : 1.0f)
```
Comment (slipher) [inline src/sgame/sg_bot_nav.cpp:438]: I think `if (sideways) { ... }` would be more readable. Also it should be "sideways" not "sideway".

LGTM

Comment (necessarily-equal) [reply src/sgame/sg_bot_nav.cpp]: updated

**Comment by bmorel** [comment]
> The two typicals problems are vents that are too tight to navigate

A proper fix to that would be something like that: https://github.com/bmorel/Unvanquished/commit/288a7f5aab286a235b5f71e8f9a34264685bf1b6 which, between other changes, checks for STEPSIZE.

**Pull request merged**